bioimageio/spec/latest/build_spec.py

```diff
@@ -412,3 +413,31 @@ def build_spec(
     model = spec.schema.Model().load(serialized)
 
     return model
+
+
+def add_weights(
+    model,
+    weight_uri: str,
+    root: Optional[str] = None,
+    weight_type: Optional[str] = None,
+    **weight_kwargs
+):
+    """ Add weight entry to bioimage.io model.
+    """
+    new_weights = _get_weights(weight_uri, weight_type, None, root, **weight_kwargs)[0]
+    model.weights.update(new_weights)
+
+    serialized = spec.schema.Model().dump(model)
+    model = spec.schema.Model().load(serialized)
+
+    return model
+
+
+def serialize_spec(model, out_path, clear_defaults=True):
```

Comment:
now, instead of potentially adding default values we potentially remove explicitly set default values... 

I would prefer explicit over implicit and in doubt write out the defaults instead of removing anything...
Maybe we can find a better way forward in https://github.com/bioimage-io/spec-bioimage-io/issues/118 first